app/uk/gov/hmrc/claimvatenrolmentfrontend/services/ClaimVatEnrolmentService.scala

```diff
@@ -34,6 +34,61 @@ class ClaimVatEnrolmentService @Inject()(auditConnector: AuditConnector,
                                          allocateEnrolmentService: AllocateEnrolmentService,
                                          journeyService: JourneyService) {
 
+  def claimVatEnrolment(credentialId: String,
+                        groupId: String,
+                        internalId: String,
+                        journeyId: String
+                       )(implicit hc: HeaderCarrier,
+                         request: Request[_],
+                         ec: ExecutionContext): Future[ClaimVatEnrolmentResponse] = {
+    journeyService.retrieveJourneyData(journeyId, internalId).flatMap {
+      journeyData =>
+        allocateEnrolmentService.allocateEnrolment(journeyData, credentialId, groupId).flatMap {
+          case EnrolmentSuccess =>
+            sendAuditEvent(journeyData, isSuccessful = true)
+            journeyService.retrieveJourneyConfig(journeyId, internalId).map {
+              journeyConfig =>
+                Right(journeyConfig.continueUrl)
+            }
+          case MultipleEnrolmentsInvalid =>
+            sendAuditEvent(journeyData, isSuccessful = false, Some(MultipleEnrolmentsInvalid.message))
+            Future.successful(Left(CannotAssignMultipleMtdvatEnrolments))
+          case InvalidKnownFacts =>
+            callEnrolmentStoreProxy(journeyData, UsersFound.message, InvalidKnownFacts.message)
+          case EnrolmentFailure(_) =>
+            callEnrolmentStoreProxy(journeyData, UsersFound.message, NoUsersFound.message, enrolmentFailure = true)
+        }
+    }
+  }
+
+  private def callEnrolmentStoreProxy(journeyData: VatKnownFacts,
+                                      usersFoundMessage: String,
+                                      noUsersFoundMessage: String,
+                                      enrolmentFailure: Boolean = false
+                                     )(implicit hc: HeaderCarrier,
+                                      request: Request[_],
+                                      ec: ExecutionContext): Future[ClaimVatEnrolmentResponse] = {
+    allocateEnrolmentService.getUserIds(journeyData.vatNumber).map {
+      case NoUsersFound if enrolmentFailure =>
+        sendAuditEvent(journeyData, isSuccessful = false, Some(noUsersFoundMessage))
+        throw new InternalServerException(NoUsersFound.message)
+      case NoUsersFound =>
+        sendAuditEvent(journeyData, isSuccessful = false, Some(noUsersFoundMessage))
```

Comment:
Should this audit be the known facts mismatch one sendAuditEvent(journeyData, isSuccessful = false, Some(InvalidKnownFacts.message))

Reply:
message for this is passed in this case:
case InvalidKnownFacts =>
            callEnrolmentStoreProxy(journeyData, UsersFound.message, InvalidKnownFacts.message)
so should be as you expect, but I will do small refactor to be more explicit about that

Reply:
done